packages/respect-core/src/modules/__tests__/flow-runner/call-api-and-analyze-results.test.ts

```diff
@@ -7,12 +6,20 @@ import {
 } from '../../flow-runner/index.js';
 import { ApiFetcher } from '../../../utils/api-fetcher.js';
 
-vi.mock('undici');
+const originalFetch = global.fetch;
+
+beforeAll(() => {
+  // Reset fetch mock before each test
+  global.fetch = vi.fn();
+});
+
+afterAll(() => {
+  // Restore original fetch after each test
```

Comment:
We already do mocks resetting (see Vite config). Also, please remove redundant comments.

Reply:
BTW, can we pass a mocked fetch instead of mocking the real one?